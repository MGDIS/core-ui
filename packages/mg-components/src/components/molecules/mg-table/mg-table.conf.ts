import type { Size } from '../../../types';
/**
 * Sizes
 */
// Define possible sizes
export const sizes = ['small', 'medium', 'large', 'xlarge'] as const;
// Define Table size type and ensure it is a valid size
export type TableSizeType = Size & (typeof sizes)[number];
/**
 * Columns
 */
// Define possible text alignments
export const textAlignments = ['left', 'center', 'right'] as const;
// Define possible data types
export const dataTypes = ['string', 'numeric', 'date'] as const;
// Define possible data types
export type datatypesType = (typeof dataTypes)[number];
// Define columns type
export type ColumnsType = {
  [key: number]: { align?: (typeof textAlignments)[number]; sortable?: boolean; datatype?: datatypesType };
};
