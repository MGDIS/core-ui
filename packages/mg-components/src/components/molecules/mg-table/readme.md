

<!-- Auto Generated Below -->


## Properties

| Property    | Attribute    | Description                             | Type                                                                                                                             | Default     |
| ----------- | ------------ | --------------------------------------- | -------------------------------------------------------------------------------------------------------------------------------- | ----------- |
| `columns`   | --           | Define column properties                | `{ [key: number]: { align?: "center" \| "right" \| "left"; sortable?: boolean; datatype?: "string" \| "numeric" \| "date"; }; }` | `undefined` |
| `fullWidth` | `full-width` | Define if table fits its parent element | `boolean`                                                                                                                        | `false`     |
| `size`      | `size`       | Define table size                       | `"large" \| "medium" \| "small" \| "xlarge"`                                                                                     | `'medium'`  |


## Slots

| Slot | Description   |
| ---- | ------------- |
|      | Table content |


## Dependencies

### Depends on

- [mg-icon](../../atoms/mg-icon)

### Graph
```mermaid
graph TD;
  mg-table --> mg-icon
  style mg-table fill:#f9f,stroke:#333,stroke-width:4px
```

----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
