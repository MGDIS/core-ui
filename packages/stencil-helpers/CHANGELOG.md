# @mgdis/stencil-helpers

## 1.1.0

### Minor Changes

- a4091b3: Provide autocompletion

### Patch Changes

- fbdf99a: Update dependencies

## 1.0.4

### Patch Changes

- d6a9e03: Update dependencies

## 1.0.3

### Patch Changes

- 7a7e0ec: Update dependencies

## 1.0.2

### Patch Changes

- c87d711: Update dependencies

## 1.0.1

### Patch Changes

- eb3b1ab: Update dependencies

## 1.0.0

### Major Changes

- 50174ed: Stencil helpers package release

## 1.0.0

### Major Changes

- 50174ed: Stencil helpers package release
