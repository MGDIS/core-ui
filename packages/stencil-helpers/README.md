# @mgdis/stencil-helpers

This package centralizes helper functions for your [Stencil](https://stenciljs.com/) projects. It is divided into four thematic sections:

- **Components**: Centralizes all component utilities.
- **Locale**: Centralizes all necessary locale utilities for managing content translation.
- **Storybook**: Centralizes all Storybook helpers useful for building your own Storybook.
- **Test**: Centralizes all test utilities to simplify testing your components.

## Installation

```bash
pnpm i @mgdis/stencil-helpers
```

## Documentation

Read the full [documentation](./doc/README.md)
