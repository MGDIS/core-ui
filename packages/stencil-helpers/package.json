{
  "name": "@mgdis/stencil-helpers",
  "version": "3.2.1",
  "license": "ISC",
  "author": "Nicolas Jouanno <jouanno-n@mgdis.fr>",
  "maintainers": [
    "Simon Duhem <duhem-s@mgdis.fr>",
    "Nicolas Jouanno <jouanno-n@mgdis.fr>"
  ],
  "files": [
    "dist"
  ],
  "main": "./dist/index.umd.js",
  "module": "./dist/index.es.js",
  "types": "dist/index.d.ts",
  "exports": {
    ".": {
      "types": "./dist/index.d.ts",
      "import": "./dist/index.es.js",
      "require": "./dist/index.umd.js"
    }
  },
  "scripts": {
    "build": "vite build",
    "doc": "typedoc",
    "test": "TZ=UTC vitest",
    "coverage": "TZ=UTC vitest run --coverage"
  },
  "dependencies": {
    "@mgdis/core-ui-helpers": "workspace:*"
  },
  "devDependencies": {
    "@core-ui/tsconfig": "workspace:*",
    "@core-ui/typedoc-config": "workspace:*",
    "@core-ui/vitest-config": "workspace:*",
    "@types/node": "22.15.29",
    "@vitest/coverage-istanbul": "3.2.1",
    "@vitest/coverage-v8": "3.2.1",
    "htmlfy": "0.7.5",
    "typedoc": "0.28.5",
    "typedoc-plugin-markdown": "4.6.4",
    "vite": "6.3.5",
    "vite-plugin-dts": "4.5.4",
    "vitest": "3.2.1"
  }
}
