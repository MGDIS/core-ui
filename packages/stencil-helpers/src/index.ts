export * from './components';
export * from './ide';
export * from './locale';
export * from './storybook';
export * from './tests';
