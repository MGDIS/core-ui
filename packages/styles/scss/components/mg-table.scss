@use '../utils/a11y';
@use '../base/typography';

.mg-c-table {
  // TODO: put variables in components custom properties after beta
  --mg-c-table-cell-spacing: var(--mg-b-size-8) var(--mg-b-size-16);
  --mg-c-table-cell-height: var(--mg-b-size-40);
  --mg-c-table-header-cell-height: calc(var(--mg-b-size-min-height) + 2 * var(--mg-b-size-8));

  overflow: hidden;
  border: var(--mg-b-size-border) solid var(--mg-b-color-neutral-30);
  border-radius: var(--mg-b-size-12);
  background: var(--mg-b-color-light);
  border-spacing: 0;
  color: var(--mg-b-color-dark);

  td,
  th {
    height: var(--mg-c-table-cell-height);
    box-sizing: border-box;
    padding: var(--mg-c-table-cell-spacing);
    text-align: start;
  }

  thead {
    height: var(--mg-c-table-header-cell-height);
  }

  thead,
  tfoot {
    background-color: var(--mg-b-color-neutral-10);
  }

  // Font weight for headers
  thead td,
  tfoot td,
  th {
    font-weight: 600;
  }

  tr {
    th,
    td {
      border-block-end: var(--mg-b-size-border) solid var(--mg-b-color-neutral-30);
    }
  }

  tfoot tr:last-of-type {
    th,
    td {
      border-block-end: none;
    }
  }

  /**
  * Sizes
  */
  &.mg-c-table--size-small {
    --mg-c-table-cell-spacing: var(--mg-b-size-4) var(--mg-b-size-16);
    --mg-c-table-cell-height: var(--mg-b-size-32);
    --mg-c-table-header-cell-height: var(--mg-b-size-32);
  }

  &.mg-c-table--size-large {
    --mg-c-table-cell-height: var(--mg-c-table-header-cell-height);
  }

  &.mg-c-table--size-xlarge {
    --mg-c-table-cell-spacing: var(--mg-b-size-12) var(--mg-b-size-16);
    --mg-c-table-cell-height: var(--mg-b-size-64);
    --mg-c-table-header-cell-height: var(--mg-b-size-64);
  }

  /**
  * full width
  */
  &.mg-c-table--full-width {
    width: 100%;
  }

  &-wrapper {
    overflow: auto;
  }

  /**
  * Sort
  */
  /* stylelint-disable no-descending-specificity -- keeping this down the file */
  thead td,
  thead th {
    position: relative;

    .mg-c-table__sort-button {
      display: inline-flex;
      align-items: center;
      padding: 0;
      border: none;
      margin: 0;
      background: none;
      color: inherit;
      font-size: inherit;
      font-weight: 600;
      gap: var(--mg-b-size-4);
      text-align: left;

      // Extend click area
      &::after {
        position: absolute;
        content: '';
        inset: 0;
      }

      &:hover {
        cursor: pointer;
      }
    }

    &:has(.mg-c-table__sort-button):hover,
    &[aria-sort='descending'],
    &[aria-sort='ascending'] {
      background-color: var(--mg-b-color-neutral-20);
    }
  }
}
