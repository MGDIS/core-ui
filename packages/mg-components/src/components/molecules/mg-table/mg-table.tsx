import { Component, Element, State, Prop, Watch } from '@stencil/core';
import { ClassList, toString, createID, isValidString } from '@mgdis/stencil-helpers';
import { type TableSizeType, type ColumnsAlignmentType, alignments, sizes } from './mg-table.conf';

/**
 * Check if value is a valid align object
 * @param newValue - New value
 * @returns True if value is a valid align object
 */
const isValidAlignObject = (newValue: unknown): boolean =>
  typeof newValue === 'object' &&
  !Array.isArray(newValue) &&
  Object.entries(newValue).every(([key, value]) => {
    return !isNaN(parseInt(key)) && alignments.includes(value);
  });

/**
 * @slot - Table content
 */
@Component({
  tag: 'mg-table',
  styleUrl: '../../../../node_modules/@mgdis/styles/dist/components/mg-table.css',
  shadow: true,
})
export class MgTable {
  /************
   * Internal *
   ************/

  // Classes
  private readonly componentClass = 'mg-c-table';
  private readonly componentWrapperClass = 'mg-c-table-wrapper';
  private readonly classFullWidth = 'mg-c-table--full-width';

  // Alignment stylesheet
  private alignmentStylesheet = new CSSStyleSheet();

  /**************
   * Decorators *
   **************/

  /**
   * Get component DOM element
   */
  @Element() element: HTMLMgTableElement;

  /**
   * Define table size
   */
  @Prop() size: TableSizeType = 'medium';
  @Watch('size')
  watchSize(newValue: MgTable['size'], oldValue?: MgTable['size']): void {
    if (!sizes.includes(newValue)) {
      throw new Error(`<mg-table> prop "size" must be one of: ${sizes.join(', ')}. Passed value: ${toString(newValue)}.`);
    } else {
      if (oldValue !== undefined) {
        this.classCollection.delete(`mg-c-table--size-${oldValue}`);
      }
      this.classCollection.add(`mg-c-table--size-${newValue}`);
    }
  }

  /**
   * Define if table fits its parent element
   */
  @Prop() fullWidth = false;
  @Watch('fullWidth')
  watchFullWidth(newValue: MgTable['fullWidth']): void {
    if (newValue) {
      this.classCollection.add(this.classFullWidth);
    } else {
      this.classCollection.delete(this.classFullWidth);
    }
  }

  /**
   * Define column alignment.
   *
   * Can be a string: `left`, `center`, `right`; In this case all columns will have the same alignment.
   *
   * Can be an array: `['left', 'center', 'right']`; In this case each column will have the corresponding alignment.
   *
   * Can be an object: `{ 2: 'center' }`; In this case the column 2 will have the corresponding alignment.
   */
  @Prop() columnsAlignment: ColumnsAlignmentType;
  @Watch('columnsAlignment')
  watchColumnAlign(newValue: MgTable['columnsAlignment']): void {
    if (newValue !== undefined) {
      // String value
      if (typeof newValue === 'string' && alignments.includes(newValue)) {
        this.classCollection.add(`mg-c-table--align-${this.columnsAlignment}`);
      }
      // Object value
      else if (typeof newValue === 'object' && isValidAlignObject(newValue)) {
        Object.entries(newValue).forEach(([key, value]) => {
          this.alignmentStylesheet.insertRule(`.${this.componentClass}{th:nth-child(${key}),td:nth-child(${key}){text-align:${value}}}`);
        });
      } else {
        throw new Error(`<mg-table> prop "columnsAlignment" can be a string or an Object, values must be one of ${alignments.join(', ')}. Passed value: ${toString(newValue)}.`);
      }
    }
  }

  /**
   * Class collection
   */
  @State() classCollection: ClassList = new ClassList([this.componentClass]);

  /**
   * Render Table
   * @param table - Table element
   */
  private renderTable(table: HTMLTableElement): void {
    // Remove table classes
    table.className = '';
    // Add table classes
    table.classList.add(...this.classCollection.classes);

    // Add alignment to columns
    this.element.shadowRoot.adoptedStyleSheets = [...this.element.shadowRoot.adoptedStyleSheets, this.alignmentStylesheet];

    // Create div wrapper for responsive behavior
    const divWrapper = document.createElement('div');
    divWrapper.setAttribute('class', this.componentWrapperClass);
    divWrapper.setAttribute('tabindex', '0');
    divWrapper.setAttribute('role', 'region');

    // Get table title
    const captionTable = table.querySelector('caption');

    if (captionTable !== null) {
      // Ensure caption is not visible (otherwhise the table does not have rounded corners)
      captionTable.className = 'mg-u-visually-hidden';
      // Link caption to div
      if (!isValidString(captionTable.id)) {
        captionTable.setAttribute('id', createID(`${this.componentClass}-caption`));
      }
      divWrapper.setAttribute('aria-labelledby', captionTable.id);
    }
    divWrapper.append(table);

    // Remove existing table from shadow root
    const existingTable = this.element.shadowRoot.querySelector(`.${this.componentWrapperClass}`);
    if (existingTable !== null) {
      existingTable.remove();
    }
    // Append table
    this.element.shadowRoot.append(divWrapper);
  }

  /*************
   * Lifecycle *
   *************/

  /**
   * Set variables and validate props
   */
  componentDidLoad(): void {
    this.watchSize(this.size);
    this.watchFullWidth(this.fullWidth);
    this.watchColumnAlign(this.columnsAlignment);
    this.renderTable(this.element.querySelector('table'));
  }

  /**
   * Update table
   */
  componentWillUpdate(): void {
    this.renderTable(this.element.shadowRoot.querySelector('table'));
  }
}
