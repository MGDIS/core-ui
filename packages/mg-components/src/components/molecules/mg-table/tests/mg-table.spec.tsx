import { h } from '@stencil/core';
import { newSpecPage } from '@stencil/core/testing';
import { MgTable } from '../mg-table';
import { sizes, alignments } from '../mg-table.conf';
import { toString } from '@mgdis/stencil-helpers';

const getPage = (args, caption?) =>
  newSpecPage({
    components: [MgTable],
    template: () => (
      <mg-table {...args}>
        <table>
          {caption}
          <thead>
            <tr>
              <th>Dev</th>
              <th>Role</th>
              <th>Test signature</th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <th>Simon</th>
              <td>Archi</td>
              <td>Blu / Daron Crew</td>
            </tr>
            <tr>
              <th>Nico</th>
              <td>Dev front-end</td>
              <td>DC Comics (Batman, Jocker, etc.)</td>
            </tr>
            <tr>
              <th>Guirec</th>
              <td>Dev front-end</td>
              <td>Pat'Patrouille</td>
            </tr>
          </tbody>
        </table>
      </mg-table>
    ),
  });

describe('mg-table', () => {
  test.each([
    {},
    { size: 'small' },
    { size: 'large' },
    { size: 'xlarge' },
    { fullWidth: true },
    { columnsAlignment: 'left' },
    { columnsAlignment: 'center' },
    { columnsAlignment: 'right' },
    { columnsAlignment: { 2: 'center' } },
  ])('Should render with args %o:', async args => {
    const { root } = await getPage(args);
    expect(root).toMatchSnapshot();
  });

  test.each(['', ' '])('Should not render with invalid size property: %s', async size => {
    expect.assertions(1);
    try {
      await getPage({ size });
    } catch (err) {
      expect(err.message).toEqual(`<mg-table> prop "size" must be one of: ${sizes.join(', ')}. Passed value: ${size}.`);
    }
  });

  test.each(['', ' ', 'blu', ['center', 'left'], { a: 'left' }, { 2: 'blu' }])('Should not render with invalid columnsAlignment property: %s', async columnsAlignment => {
    expect.assertions(1);
    try {
      await getPage({ columnsAlignment });
    } catch (err) {
      expect(err.message).toEqual(
        `<mg-table> prop "columnsAlignment" can be a string or an Object, values must be one of ${alignments.join(', ')}. Passed value: ${toString(columnsAlignment)}.`,
      );
    }
  });

  test('Should update mg-table classes on props change', async () => {
    const page = await getPage({ size: 'small' });
    const mgTable = page.doc.querySelector('mg-table');

    expect(page.root).toMatchSnapshot();

    // Change size
    mgTable.size = 'xlarge';
    await page.waitForChanges();

    expect(page.root).toMatchSnapshot();

    // Change size
    mgTable.fullWidth = true;
    await page.waitForChanges();

    expect(page.root).toMatchSnapshot();
  });

  test('Should link table to caption', async () => {
    const caption = <caption id="blu">Test caption</caption>;
    const { root } = await getPage({}, caption);
    expect(root).toMatchSnapshot();
  });

  test('Should add caption id if not set', async () => {
    const caption = <caption>Test caption</caption>;
    const page = await getPage({}, caption);

    // Get the caption element from the document
    const captionElement = page.doc.querySelector('mg-table').shadowRoot.querySelector('caption');

    // Ensure an id has been added
    expect(captionElement).toHaveAttribute('id');
    const generatedId = captionElement.getAttribute('id');

    // Define the expected ID format
    const idRegex = /^mg-c-table-caption-[a-z0-9]{10}$/;
    expect(generatedId).toMatch(idRegex);

    // Ensure the wrapper div references the caption id
    const tableWrapper = page.doc.querySelector('mg-table').shadowRoot.querySelector('div');
    expect(tableWrapper).toHaveAttribute('aria-labelledby', generatedId);
  });
});
