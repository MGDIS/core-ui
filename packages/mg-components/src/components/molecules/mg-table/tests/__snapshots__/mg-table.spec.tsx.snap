// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mg-table Should link table to caption 1`] = `
<mg-table>
  <template shadowrootmode="open">
    <div aria-labelledby="blu" class="mg-c-table-wrapper" role="region" tabindex="0">
      <table class="mg-c-table mg-c-table--size-medium">
        <caption class="mg-u-visually-hidden" id="blu">
          Test caption
        </caption>
        <thead>
          <tr>
            <th>
              Dev
            </th>
            <th>
              Role
            </th>
            <th>
              Test signature
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <th>
              Simon
            </th>
            <td>
              Archi
            </td>
            <td>
              Blu / Daron Crew
            </td>
          </tr>
          <tr>
            <th>
              Nico
            </th>
            <td>
              Dev front-end
            </td>
            <td>
              DC Comics (Batman, Jocker, etc.)
            </td>
          </tr>
          <tr>
            <th>
              Guirec
            </th>
            <td>
              Dev front-end
            </td>
            <td>
              Pat'Patrouille
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </template>
</mg-table>
`;

exports[`mg-table Should render with args { columnsAlignment: 'center' }: 1`] = `
<mg-table>
  <template shadowrootmode="open">
    <div class="mg-c-table-wrapper" role="region" tabindex="0">
      <table class="mg-c-table mg-c-table--align-center mg-c-table--size-medium">
        <thead>
          <tr>
            <th>
              Dev
            </th>
            <th>
              Role
            </th>
            <th>
              Test signature
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <th>
              Simon
            </th>
            <td>
              Archi
            </td>
            <td>
              Blu / Daron Crew
            </td>
          </tr>
          <tr>
            <th>
              Nico
            </th>
            <td>
              Dev front-end
            </td>
            <td>
              DC Comics (Batman, Jocker, etc.)
            </td>
          </tr>
          <tr>
            <th>
              Guirec
            </th>
            <td>
              Dev front-end
            </td>
            <td>
              Pat'Patrouille
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </template>
</mg-table>
`;

exports[`mg-table Should render with args { columnsAlignment: 'left' }: 1`] = `
<mg-table>
  <template shadowrootmode="open">
    <div class="mg-c-table-wrapper" role="region" tabindex="0">
      <table class="mg-c-table mg-c-table--align-left mg-c-table--size-medium">
        <thead>
          <tr>
            <th>
              Dev
            </th>
            <th>
              Role
            </th>
            <th>
              Test signature
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <th>
              Simon
            </th>
            <td>
              Archi
            </td>
            <td>
              Blu / Daron Crew
            </td>
          </tr>
          <tr>
            <th>
              Nico
            </th>
            <td>
              Dev front-end
            </td>
            <td>
              DC Comics (Batman, Jocker, etc.)
            </td>
          </tr>
          <tr>
            <th>
              Guirec
            </th>
            <td>
              Dev front-end
            </td>
            <td>
              Pat'Patrouille
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </template>
</mg-table>
`;

exports[`mg-table Should render with args { columnsAlignment: 'right' }: 1`] = `
<mg-table>
  <template shadowrootmode="open">
    <div class="mg-c-table-wrapper" role="region" tabindex="0">
      <table class="mg-c-table mg-c-table--align-right mg-c-table--size-medium">
        <thead>
          <tr>
            <th>
              Dev
            </th>
            <th>
              Role
            </th>
            <th>
              Test signature
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <th>
              Simon
            </th>
            <td>
              Archi
            </td>
            <td>
              Blu / Daron Crew
            </td>
          </tr>
          <tr>
            <th>
              Nico
            </th>
            <td>
              Dev front-end
            </td>
            <td>
              DC Comics (Batman, Jocker, etc.)
            </td>
          </tr>
          <tr>
            <th>
              Guirec
            </th>
            <td>
              Dev front-end
            </td>
            <td>
              Pat'Patrouille
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </template>
</mg-table>
`;

exports[`mg-table Should render with args { columnsAlignment: { '2': 'center' } }: 1`] = `
<mg-table>
  <template shadowrootmode="open">
    <div class="mg-c-table-wrapper" role="region" tabindex="0">
      <table class="mg-c-table mg-c-table--size-medium">
        <thead>
          <tr>
            <th>
              Dev
            </th>
            <th>
              Role
            </th>
            <th>
              Test signature
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <th>
              Simon
            </th>
            <td>
              Archi
            </td>
            <td>
              Blu / Daron Crew
            </td>
          </tr>
          <tr>
            <th>
              Nico
            </th>
            <td>
              Dev front-end
            </td>
            <td>
              DC Comics (Batman, Jocker, etc.)
            </td>
          </tr>
          <tr>
            <th>
              Guirec
            </th>
            <td>
              Dev front-end
            </td>
            <td>
              Pat'Patrouille
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </template>
</mg-table>
`;

exports[`mg-table Should render with args { fullWidth: true }: 1`] = `
<mg-table>
  <template shadowrootmode="open">
    <div class="mg-c-table-wrapper" role="region" tabindex="0">
      <table class="mg-c-table mg-c-table--full-width mg-c-table--size-medium">
        <thead>
          <tr>
            <th>
              Dev
            </th>
            <th>
              Role
            </th>
            <th>
              Test signature
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <th>
              Simon
            </th>
            <td>
              Archi
            </td>
            <td>
              Blu / Daron Crew
            </td>
          </tr>
          <tr>
            <th>
              Nico
            </th>
            <td>
              Dev front-end
            </td>
            <td>
              DC Comics (Batman, Jocker, etc.)
            </td>
          </tr>
          <tr>
            <th>
              Guirec
            </th>
            <td>
              Dev front-end
            </td>
            <td>
              Pat'Patrouille
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </template>
</mg-table>
`;

exports[`mg-table Should render with args { size: 'large' }: 1`] = `
<mg-table>
  <template shadowrootmode="open">
    <div class="mg-c-table-wrapper" role="region" tabindex="0">
      <table class="mg-c-table mg-c-table--size-large">
        <thead>
          <tr>
            <th>
              Dev
            </th>
            <th>
              Role
            </th>
            <th>
              Test signature
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <th>
              Simon
            </th>
            <td>
              Archi
            </td>
            <td>
              Blu / Daron Crew
            </td>
          </tr>
          <tr>
            <th>
              Nico
            </th>
            <td>
              Dev front-end
            </td>
            <td>
              DC Comics (Batman, Jocker, etc.)
            </td>
          </tr>
          <tr>
            <th>
              Guirec
            </th>
            <td>
              Dev front-end
            </td>
            <td>
              Pat'Patrouille
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </template>
</mg-table>
`;

exports[`mg-table Should render with args { size: 'small' }: 1`] = `
<mg-table>
  <template shadowrootmode="open">
    <div class="mg-c-table-wrapper" role="region" tabindex="0">
      <table class="mg-c-table mg-c-table--size-small">
        <thead>
          <tr>
            <th>
              Dev
            </th>
            <th>
              Role
            </th>
            <th>
              Test signature
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <th>
              Simon
            </th>
            <td>
              Archi
            </td>
            <td>
              Blu / Daron Crew
            </td>
          </tr>
          <tr>
            <th>
              Nico
            </th>
            <td>
              Dev front-end
            </td>
            <td>
              DC Comics (Batman, Jocker, etc.)
            </td>
          </tr>
          <tr>
            <th>
              Guirec
            </th>
            <td>
              Dev front-end
            </td>
            <td>
              Pat'Patrouille
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </template>
</mg-table>
`;

exports[`mg-table Should render with args { size: 'xlarge' }: 1`] = `
<mg-table>
  <template shadowrootmode="open">
    <div class="mg-c-table-wrapper" role="region" tabindex="0">
      <table class="mg-c-table mg-c-table--size-xlarge">
        <thead>
          <tr>
            <th>
              Dev
            </th>
            <th>
              Role
            </th>
            <th>
              Test signature
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <th>
              Simon
            </th>
            <td>
              Archi
            </td>
            <td>
              Blu / Daron Crew
            </td>
          </tr>
          <tr>
            <th>
              Nico
            </th>
            <td>
              Dev front-end
            </td>
            <td>
              DC Comics (Batman, Jocker, etc.)
            </td>
          </tr>
          <tr>
            <th>
              Guirec
            </th>
            <td>
              Dev front-end
            </td>
            <td>
              Pat'Patrouille
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </template>
</mg-table>
`;

exports[`mg-table Should render with args {}: 1`] = `
<mg-table>
  <template shadowrootmode="open">
    <div class="mg-c-table-wrapper" role="region" tabindex="0">
      <table class="mg-c-table mg-c-table--size-medium">
        <thead>
          <tr>
            <th>
              Dev
            </th>
            <th>
              Role
            </th>
            <th>
              Test signature
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <th>
              Simon
            </th>
            <td>
              Archi
            </td>
            <td>
              Blu / Daron Crew
            </td>
          </tr>
          <tr>
            <th>
              Nico
            </th>
            <td>
              Dev front-end
            </td>
            <td>
              DC Comics (Batman, Jocker, etc.)
            </td>
          </tr>
          <tr>
            <th>
              Guirec
            </th>
            <td>
              Dev front-end
            </td>
            <td>
              Pat'Patrouille
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </template>
</mg-table>
`;

exports[`mg-table Should update mg-table classes on props change 1`] = `
<mg-table>
  <template shadowrootmode="open">
    <div class="mg-c-table-wrapper" role="region" tabindex="0">
      <table class="mg-c-table mg-c-table--size-small">
        <thead>
          <tr>
            <th>
              Dev
            </th>
            <th>
              Role
            </th>
            <th>
              Test signature
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <th>
              Simon
            </th>
            <td>
              Archi
            </td>
            <td>
              Blu / Daron Crew
            </td>
          </tr>
          <tr>
            <th>
              Nico
            </th>
            <td>
              Dev front-end
            </td>
            <td>
              DC Comics (Batman, Jocker, etc.)
            </td>
          </tr>
          <tr>
            <th>
              Guirec
            </th>
            <td>
              Dev front-end
            </td>
            <td>
              Pat'Patrouille
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </template>
</mg-table>
`;

exports[`mg-table Should update mg-table classes on props change 2`] = `
<mg-table>
  <template shadowrootmode="open">
    <div class="mg-c-table-wrapper" role="region" tabindex="0">
      <table class="mg-c-table mg-c-table--size-xlarge">
        <thead>
          <tr>
            <th>
              Dev
            </th>
            <th>
              Role
            </th>
            <th>
              Test signature
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <th>
              Simon
            </th>
            <td>
              Archi
            </td>
            <td>
              Blu / Daron Crew
            </td>
          </tr>
          <tr>
            <th>
              Nico
            </th>
            <td>
              Dev front-end
            </td>
            <td>
              DC Comics (Batman, Jocker, etc.)
            </td>
          </tr>
          <tr>
            <th>
              Guirec
            </th>
            <td>
              Dev front-end
            </td>
            <td>
              Pat'Patrouille
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </template>
</mg-table>
`;

exports[`mg-table Should update mg-table classes on props change 3`] = `
<mg-table>
  <template shadowrootmode="open">
    <div class="mg-c-table-wrapper" role="region" tabindex="0">
      <table class="mg-c-table mg-c-table--full-width mg-c-table--size-xlarge">
        <thead>
          <tr>
            <th>
              Dev
            </th>
            <th>
              Role
            </th>
            <th>
              Test signature
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <th>
              Simon
            </th>
            <td>
              Archi
            </td>
            <td>
              Blu / Daron Crew
            </td>
          </tr>
          <tr>
            <th>
              Nico
            </th>
            <td>
              Dev front-end
            </td>
            <td>
              DC Comics (Batman, Jocker, etc.)
            </td>
          </tr>
          <tr>
            <th>
              Guirec
            </th>
            <td>
              Dev front-end
            </td>
            <td>
              Pat'Patrouille
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </template>
</mg-table>
`;
